pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go

```diff
@@ -113,3 +125,99 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 		dswp.desiredStateOfWorld.DeletePod(dswPodUID, dswPodToAdd.VolumeName, dswPodToAdd.NodeName)
 	}
 }
+
+func (dswp *desiredStateOfWorldPopulator) isPodTerminated(pod *v1.Pod) bool {
+	return pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodSucceeded || (pod.DeletionTimestamp != nil && notRunning(pod.Status.ContainerStatuses))
+}
+
+// notRunning returns true if every status is terminated or waiting, or the status list
+// is empty.
+func notRunning(statuses []v1.ContainerStatus) bool {
+	for _, status := range statuses {
+		if status.State.Terminated == nil && status.State.Waiting == nil {
+			return false
+		}
+	}
+	return true
+}
+
+func (dswp *desiredStateOfWorldPopulator) findAndAddActivePods() {
```

Comment:
This method is mostly a duplicate of `processPodVolumes(...)`. Let's avoid the duplication and move the common logic to `attachdetach/util/...`. Also, now that I look at it, you don't really need `desiredStateOfWorld.PodExists(...)` at all. since `desiredStateOfWorld.AddPod(...)` is a no-op if the specified pod/volume/node already exists.